common/lib/dependabot/pull_request_creator/message_builder.rb

```diff
@@ -357,14 +372,17 @@ def dependency_set
       end
 
       def dependency_links
-        dependencies.map do |dependency|
-          if source_url(dependency)
-            "[#{dependency.display_name}](#{source_url(dependency)})"
-          elsif homepage_url(dependency)
-            "[#{dependency.display_name}](#{homepage_url(dependency)})"
-          else
-            dependency.display_name
-          end
+        uniq_deps = dependencies.each_with_object({}) { |dep, memo| memo[dep.name] ||= dep }.values
+        uniq_deps.map { |dep| dependency_link(dep) }
```

Comment:
Not sure if any of the work done here is potentially expensive, but seems like we could memoize this. If it's always a small collection and the work is not expensive then keeping it as is seems fine 👍